images/image.go

```diff
@@ -387,6 +382,49 @@ func Children(ctx context.Context, provider content.Provider, desc ocispec.Descr
 	return descs, nil
 }
 
+const readValidatedManifestBlobRetries = 3
+const readValidatedManifestBlobDelaySeconds = 5
+
+func readValidatedManifestBlobWithRetry(ctx context.Context, provider content.Provider, desc ocispec.Descriptor) ([]byte, error) {
+
+	retriesRemaining := readValidatedManifestBlobRetries
+
+	for {
+
+		blob, err := readValidatedManifestBlob(ctx, provider, desc)
+		if err == nil {
+			return blob, nil
+		}
+
+		if retriesRemaining > 0 {
+			log.G(ctx).
+				WithField("retries-remaining", retriesRemaining).
+				WithField("retry-delay-seconds", readValidatedManifestBlobDelaySeconds).
+				WithError(err).
+				Warn("readValidatedManifestBlob failed; retrying")
+
+			retriesRemaining -= 1
+
+			time.Sleep(time.Duration(readValidatedManifestBlobDelaySeconds) * time.Second)
```

Comment:
This should be fine, since `time.Sleep` is meant to interact safely with goroutines (which is what this is called within).